### PR TITLE
#38 Split into multiple files

### PR DESCRIPTION
**Pull request by kyoungho**

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -144,7 +144,7 @@ issues:
       text: "Error return value of `connector.Delete` is not checked"
 
     # Ignore complicated function that gocritic is suspecting
-    - path: rticonnextdds_connector.go
+    - path: .go
```
Comment (metarsit) [inline .golangci.yaml:147]: what is this? We should _path_ it to that very file

File: info.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+/********************
+* Private Functions *
+********************/
+
+func newInfos(input *Input) *Infos {
+	// Error checking for the input is skipped because it was already checked
+	return &Infos{
+		input: input,
+	}
+}
```
Comment (metarsit) [inline info.go:46]: Let's put private function(s) at the bottom

File: input.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		return nil, errors.New("invalid Subscription::DataReader name")
+	}
+	input.name = inputName
+	input.Samples = newSamples(input)
+	input.Infos = newInfos(input)
+
+	connector.Inputs = append(connector.Inputs, *input)
+
+	return input, nil
+}
```
Comment (metarsit) [inline input.go:63]: Let's put private function(s) at the bottom

File: instance.go [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+/********************
+* Private Functions *
+********************/
+
+func newInstance(output *Output) *Instance {
+	// Error checking for the output is skipped because it was already checked
+	return &Instance{
+		output: output,
+	}
+}
```
Comment (metarsit) [inline instance.go:44]: Let's put private function(s) at the bottom

File: output.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	if output.native == nil {
+		return nil, errors.New("invalid Publication::DataWriter name")
+	}
+	output.name = outputName
+	output.Instance = newInstance(output)
+
+	connector.Outputs = append(connector.Outputs, *output)
+
+	return output, nil
+}
```
Comment (metarsit) [inline output.go:62]: Let's put private function(s) at the bottom

File: sample.go [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+/********************
+* Private Functions *
+********************/
+
+func newSamples(input *Input) *Samples {
+	// Error checking for the input is skipped because it was already checked
+	return &Samples{
+		input: input,
+	}
+}
```
Comment (metarsit) [inline sample.go:44]: Let's put private function(s) at the bottom

**Review by metarsit — CHANGES_REQUESTED** [review]

**Comment by metarsit** [comment]
Apart from the above comments, you should also break the test into its own files too.

i.e. `sample_test.go`

File: .golangci.yaml (hunk 1 above)
Comment (kyoungho) [reply .golangci.yaml:147]: Now, we run into this issue in multiple files and `.go` seems working for any go files. We can add multiple paths as well if it is better.

File: info.go (hunk 2 above)
Comment (kyoungho) [reply info.go]: I realized most private functions should stay in `rticonnextdds_connector.go` as they are mainly used in the functions in the file. I moved private functions to the bottom.

**Comment by kyoungho** [comment]
> Apart from the above comments, you should also break the test into its own files too.
> 
> i.e. `sample_test.go`

Well. It seems hard to break tests into multiple files as a test function like `TestDataFlow` test functions across multiple files. Maybe we should refine our test code and it has be handled by a separate issue.

File: .golangci.yaml (hunk 1 above)
Comment (metarsit) [reply .golangci.yaml:147]: Maybe it is fine, seems it is a CGO thing anyway.

**Comment by metarsit** [comment]
> > Apart from the above comments, you should also break the test into its own files too.
> > i.e. `sample_test.go`
> 
> Well. It seems hard to break tests into multiple files as a test function like `TestDataFlow` test functions across multiple files. Maybe we should refine our test code and it has be handled by a separate issue.

Let's make another ticket for that then. And maybe the test that uses multiple function, maybe we should have it as integration test?

**Review by metarsit — APPROVED** [review]

**Pull request merged**